test/coordinate/cartesian_representation.cpp

```diff
@@ -0,0 +1,270 @@
+#define BOOST_TEST_MODULE cartesian_representation_test
+
+#include <boost/test/unit_test.hpp>
+#include <boost/units/quantity.hpp>
+#include <boost/units/systems/si/plane_angle.hpp>
+#include <boost/units/systems/si/prefixes.hpp>
+#include <boost/units/systems/si/length.hpp>
+#include <boost/units/systems/angle/degrees.hpp>
+#include <boost/astronomy/coordinate/arithmetic.hpp>
+#include <boost/astronomy/coordinate/representation.hpp>
+
+using namespace std;
+using namespace boost::astronomy::coordinate;
+using namespace boost::units::si;
+using namespace boost::geometry;
+using namespace boost::units;
+namespace bud = boost::units::degree;
+
+BOOST_AUTO_TEST_SUITE(cartesian_representation_constructor)
+
+BOOST_AUTO_TEST_CASE(cartesian_quantity)
+{
+    //checking construction from value
+    auto point1 = make_cartesian_representation
+    (1.5*meter, 9.0*si::kilo*meter, 3.0*si::centi*meter);
+    BOOST_CHECK_CLOSE(point1.get_x().value(), 1.5, 0.001);
+    BOOST_CHECK_CLOSE(point1.get_y().value(), 9.0, 0.001);
+    BOOST_CHECK_CLOSE(point1.get_z().value(), 3, 0.001);
+
+    //checking whether quantity stored is as expected or not
+    BOOST_TEST((std::is_same<decltype(point1.get_x()), quantity<si::length>>::value));
+    BOOST_TEST((std::is_same<decltype(point1.get_y()), quantity<decltype(si::kilo*meter)>>::value));
+    BOOST_TEST((std::is_same<decltype(point1.get_z()), quantity<decltype(si::centi*meter)>>::value));
+
+    cartesian_representation<double, quantity<si::length>, quantity<si::length>,
+    quantity<si::length>> point2(1.5*meter, 9.0*meter, 3.0*meter);
+    BOOST_CHECK_CLOSE(point2.get_x().value(), 1.5, 0.001);
+    BOOST_CHECK_CLOSE(point2.get_y().value(), 9.0, 0.001);
+    BOOST_CHECK_CLOSE(point2.get_z().value(), 3, 0.001);
+
+    //checking whether quantity stored is as expected or not
+    BOOST_TEST((std::is_same<decltype(point2.get_x()), quantity<si::length>>::value));
+    BOOST_TEST((std::is_same<decltype(point2.get_y()), quantity<si::length>>::value));
+    BOOST_TEST((std::is_same<decltype(point2.get_z()), quantity<si::length>>::value));  
+
+    //copy constructor
+    auto point3 = make_cartesian_representation(point1);
+    BOOST_CHECK_CLOSE(point1.get_x().value(), point3.get_x().value(), 0.001);
+    BOOST_CHECK_CLOSE(point1.get_y().value(), point3.get_y().value(), 0.001);
+    BOOST_CHECK_CLOSE(point1.get_z().value(), point3.get_z().value(), 0.001);
+
+    //checking whether quantity stored is as expected or not
+    BOOST_TEST((std::is_same<decltype(point3.get_x()), quantity<si::length>>::value));
+    BOOST_TEST((std::is_same<decltype(point3.get_y()), quantity<decltype(si::kilo*meter)>>::value));
+    BOOST_TEST((std::is_same<decltype(point3.get_z()), quantity<decltype(si::centi*meter)>>::value));
+
+    cartesian_representation<double, quantity<si::length>, quantity<si::length>,
+    quantity<si::length>> point4(point1);
+    BOOST_CHECK_CLOSE(point1.get_x().value(), point4.get_x().value(), 0.001);
```

Comment:
there are some implicit conversions happening from km to m and cm to m. How values from both points can be the same?

Reply:
Actually I forgot that unit conversion is not allowed for constructors